pandaproxy/sr: bazelize compatibility_protobuf test

IoannisRP opened the pull request:
Implements: [CORE-7549](https://redpandadata.atlassian.net/browse/CORE-7549) - `compatibility_protobuf.cc`

Had to change how the known types `.proto` files are ingested. Without the changes in the `schema_registry_protos` bazel target, they were visible using the full local path from workspace. i.e.
```
import "src/v/pandaproxy/schema_registry/protobuf/google/protobuf/any.proto";
```

## Backports Required

<!-- Checking at least one of the checkboxes is REQUIRED if this PR is not a backport. -->

- [x] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [ ] v24.3.x
- [ ] v24.2.x
- [ ] v24.1.x

## Release Notes

* none


[CORE-7549]: https://redpandadata.atlassian.net/browse/CORE-7549?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

IoannisRP committed:
pandaproxy/sr: bazelize compatibility_protobuf test

BenPope commented on src/v/pandaproxy/schema_registry/protobuf/BUILD:
```diff
@@ -0,0 +1,41 @@
+load("@rules_proto//proto:defs.bzl", "proto_library")
+
+proto_library(
+    name = "schema_registry_protos",
```
nitpick: `"//src/v/pandaproxy/schema_registry/protobuf:schema_registry_protos"` is quite wordy
```suggestion
    name = "builtin",
```

IoannisRP replied:
As a follow up to this PR, we will refactor all the `.proto` files under `src/schema/protobuf`

BenPope reviewed (commented):
I wonder if it's worth moving the well-known types from `pandaproxy/BUILD` to `schema_registry/protobuf`, as well?

Or maybe into `/src/v/schema/protobuf`?

I tried to package up `confluent_proto`, `google_type_proto`, and `well_known_proto` into their own cc_libraries, but failed to get that to work.

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--5a8c4513-a237-473c-bacf-4946e048adc8-->
<!--TEST RESULTS-->
<!--a74bad67-1f48-4442-a3cb-c38890b1ca62-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/60587">build#60587</a></summary>


| test_id | test_kind | job_url | test_status | passed |
|:----------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| idempotency_tests_rpunit.idempotency_tests_rpunit | unit | https://buildkite.com/redpanda/redpanda/builds/60587#01945015-8290-4e6e-af01-e2d7f07604e5 | FLAKY | 1/2 |
| rm_stm_tests_rpunit.rm_stm_tests_rpunit | unit | https://buildkite.com/redpanda/redpanda/builds/60587#01945015-8290-4e6e-af01-e2d7f07604e5 | FLAKY | 1/2 |
| rm_stm_tests_rpunit.rm_stm_tests_rpunit | unit | https://buildkite.com/redpanda/redpanda/builds/60587#01945015-8291-4c0e-9c44-9c932242e0be | FLAKY | 1/2 |
| rptest.tests.partition_reassignments_test.PartitionReassignmentsTest.test_reassignments_kafka_cli | ducktape | https://buildkite.com/redpanda/redpanda/builds/60587#0194505e-5eb7-4e41-8fcb-5b355e830bee | FLAKY | 3/6 |
| rptest.tests.scaling_up_test.ScalingUpTest.test_scaling_up_with_recovered_topic | ducktape | https://buildkite.com/redpanda/redpanda/builds/60587#01945061-c49b-460a-9f83-c60713e5b442 | FLAKY | 5/6 |
| rptest.transactions.stream_verifier_test.StreamVerifierTest.test_simple_produce_consume_txn_with_add_node | ducktape | https://buildkite.com/redpanda/redpanda/builds/60587#0194505e-5eb6-4da2-b1c1-61e84d8db16e | FLAKY | 5/6 |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/60624">build#60624</a></summary>


| test_id                                                                                                          | test_kind   | job_url                                                                                   | test_status   | passed   |
|:-----------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.partition_force_reconfiguration_test.NodeWiseRecoveryTest.test_node_wise_recovery.dead_node_count=1 | ducktape    | https://buildkite.com/redpanda/redpanda/builds/60624#01945df6-f630-4b63-9b8a-04d8d115b348 | FLAKY         | 5/6      |
</details>
<!--/body-->

rockwotj commented on src/go/rpk/pkg/serde/embed/embed_test.go:
```diff
@@ -23,7 +24,8 @@ func TestEmbeddedFiles(t *testing.T) {
 			data, err := fs.ReadFile(redpandaProtoFS, path)
 			if err == nil {
-				redpandaMap[path] = string(data)
+				trimedPath := strings.TrimPrefix(path, "_virtual_imports/schema_registry_protos/")
```
Does the symlink thing work at all? I would love to get rid of this...

IoannisRP replied:
i am not sure where the symlink would go, tbh...
In bazel, we operate within the bazel file tree. Otherwise, we operate on the actual file tree. So, we should generate a symlink from the bazel tree (generated during the bazel steps) back to the actual folder. That would mean that the bazel has to generate the symlink? Or at least, that's when i gave up on this idea.

Honestly, i don't like anything about this pathing. Any path that starts with `../../../../..` is, at the very least, a smell :sweat_smile:.

rockwotj replied:
Ah nevermind, I didn't realize: https://github.com/golang/go/issues/44507

Basically this is a test that the two copies of protos are in sync... What if there was only one copy? We wouldn't need this test...

Go always needs this to be a subdirectory of the current module, and it can't be a symlink, so maybe this does require two copies.

I looked into this, and the right thing to do is to depend on the proto files directly, not on the proto_library (which is only supposed to be depended on by rules that generate protos, which isn't what this test is for).

I've pushed what this looks like directly to your branch (hope that's cool!), feel free to blow that commit away if you have something else in mind.

IoannisRP replied:
Your suggestion looks very good! :+1: 
Merging...

rockwotj committed:
bazel/rpk/protobuf/embed: depend on the files directly

Don't depend on the protobuf library, then to get raw `.proto` files
you have to know about how the `proto_library` rule works, instead just
create a filegroup rule for the protos that the go test depends on,
which is also consumed by the `proto_library` rule.
